onedal/linear_model/logistic_regression.py

```diff
@@ -54,7 +54,11 @@ def _get_onedal_params(self, dtype=np.float32):
             "max_iter": self.max_iter,
             "C": self.C,
             "optimizer": self.solver,
-            "result_option": (intercept + "coefficients|iterations_count"),
+            "result_option": (
+                intercept + "coefficients|iterations_count" + "|inner_iterations_count"
+                if self.solver == "newton-cg"
+                else ""
+            ),
         }
```

Comment:
Some parentheses may be needed, its hard to tell what is intended by the logic here.
My assumption is that we only want to append "|inner_iterations_count" if solver is newton cg, however I think what actually happens is result_option will be empty string if solver is not newton cg